### PR TITLE
Split IPython specific functions out of utils.path

### PR DESCRIPTION
**Pull request by takluyver**
Functions like `get_ipython_dir()` are moved to a new `IPython.paths` module. This should facilitate moving the remainder of utils.path into genutils. Wrappers are left in utils.path which will still work, but issue a deprecation warning on call.

While doing this, I noticed that the code to check for pre-0.11 config files and warn about them is still in place. I think that has served its purpose and can be retired now.

### CODE REVIEW

**Comment by Carreau** [comment]
+1

File: IPython/core/magics/test_paths.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
+from contextlib import contextmanager
```
Comment (minrk) [inline IPython/core/magics/test_paths.py]: I assume this isn't meant to be in magics

Comment (takluyver) [inline IPython/core/magics/test_paths.py]: No, it wasn't. Fixed.

File: IPython/utils/path.py [hunk 2]
```diff
@@ -443,81 +367,19 @@ def target_update(target,deps,cmd):
...
-    """
-    # import here, because profiledir also imports from utils.path
-    from IPython.core.profiledir import ProfileDir
-    try:
-        pd = ProfileDir.find_profile_dir_by_name(get_ipython_dir(), profile)
-    except Exception:
-        # will raise ProfileDirError if no such profile
-        raise IOError("Profile %r not found")
-    return filefind(filename, ['.', pd.security_dir])
+    warn("get_security_file has moved to the IPython.paths module")
```
Comment (minrk) [inline IPython/utils/path.py]: get_security_file is going to leave IPython entirely, since it's a Jupyter function.

Comment (takluyver) [inline IPython/utils/path.py]: I found we were only using this in one example, and `find_connection_file`, which is already in jupyter_client, works as well for that, so I have removed this function entirely.

**Pull request merged**